hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java

```diff
@@ -321,7 +321,9 @@ private Map<StructLike, Partition> filterNewPartitionNonExists(
               "partition will be " +
               "delete and re-create with same location");
         } else {
-          partitions.put(entry.getKey(), entry.getValue());
+          // this partition is need to alter, rather than delete
+          partitionToAlter.put(entry.getKey(), entry.getValue());
+          partitionToDelete.remove(entry.getKey());
```

Comment:
need to update hive partition fields and properties.
includes lastAccessTime, transient_lastDdlTime, totalSize, numRows, numFiles.

Reply:
> need to update hive partition fields and properties. includes lastAccessTime, transient_lastDdlTime, totalSize, numRows, numFiles.

Yes, the partition is already produced by the latest info.

Reply:
As I know, the `partition` from the `partitionToCreate` map is created with a new status.

reference:https://github.com/NetEase/arctic/blob/master/hive/src/main/java/com/netease/arctic/hive/op/UpdateHiveFiles.java#L192